dm/master/server_test.go

```diff
@@ -1474,3 +1479,43 @@ func (t *testMaster) TestFetchWorkerDDLInfo(c *check.C) {
 	}()
 	wg.Wait()
 }
+
+func (t *testMaster) TestServer(c *check.C) {
+	cfg := NewConfig()
+	c.Assert(cfg.Parse([]string{"-config=./dm-master.toml"}), check.IsNil)
+
+	s := NewServer(cfg)
+
+	masterAddr := cfg.MasterAddr
+	s.cfg.MasterAddr = ""
+	err := s.Start()
+	c.Assert(terror.ErrMasterHostPortNotValid.Equal(err), check.IsTrue)
+	s.cfg.MasterAddr = masterAddr
+
+	go func() {
+		err1 := s.Start()
+		c.Assert(err1, check.IsNil)
+	}()
+
+	c.Assert(utils.WaitSomething(30, 100*time.Millisecond, func() bool {
+		return !s.closed.Get()
+	}), check.IsTrue)
+
+	t.testHTTPInterface(c, "status")
+
+	// close
+	s.Close()
```

Comment:
add a test for starting another `Server` with the same address before `s.Close`?